### ISSUE TITLE
Prometheus Provider doesn't respect a timeout in an Analysis Template Configuration

### ISSUE THREAD

**Issue by mura-s** [pipe-cd/pipe#1244]
**What happened**:

I set a very small timeout (`1µs`) in the `analysis-template.yaml`, but Prometheus query didn't time out.

https://pipecd.dev/docs/user-guide/configuration-reference/#analysis-template-configuration

It's because Prometheus Provider always uses `defaultTimeout`.

https://github.com/pipe-cd/pipe/blob/master/pkg/app/piped/analysisprovider/metrics/prometheus/prometheus.go#L31-L67

**What you expected to happen**:

Prometheus Provider respects a timeout in the `analysis-template.yaml`, and correctly times out.

**How to reproduce it**:

Set a timeout like below, and set an `ANALYSIS` deployment  step in a pipeline configuration, then execute a deployment.

```
apiVersion: pipecd.dev/v1beta1
kind: AnalysisTemplate
spec:
  metrics:
    grpc_error_rate_percentage:
      interval: 1m
      provider: prometheus-dev
      expected:
        max: 10
      query: |
        100 - sum(
            rate(
                grpc_server_handled_total{
                  grpc_code!="OK",
                  kubernetes_namespace="{{ .K8s.Namespace }}",
                  kubernetes_pod_name=~"{{ .App.Name }}-[0-9a-zA-Z]+(-[0-9a-zA-Z]+)"
                }[{{ .Args.interval }}]
            )
        )
        /
        sum(
            rate(
                grpc_server_started_total{
                  kubernetes_namespace="{{ .K8s.Namespace }}",
                  kubernetes_pod_name=~"{{ .App.Name }}-[0-9a-zA-Z]+(-[0-9a-zA-Z]+)"
                }[{{ .Args.interval }}]
            )
        ) * 100
      timeout: 1µs
```

**Environment**:
- `piped` version: v0.9.0
- `control-plane` version: v0.9.0

**Comment by nakabonne** [comment]
@mura-s Thank you! You're right, will address it :D

**Comment by nakabonne** [comment]
@mura-s This has been addressed by @sanposhiho so will be available after v1.0.0 released (planning by 1/22)
